docs/source/using-discovery/bash.rst

```diff
@@ -0,0 +1,867 @@
+
+``ssh`` - Connect to a remote machine using Secure Shell (SSH)::
+
+   ssh <user-name>@login.discovery.neu.edu
+
+Reference: `ssh(1) manual page`_
+
+.. note::
+   The ``~`` character is shorthand for specifying the home directory of the current user, i.e., ``~`` is the same as ``$HOME``.
+
+``echo`` - Display a message or the value of a variable.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  echo "Hello, world!"
+  Hello, world!
+
+.. note:: 
+  Variable names are accessible by appending its name to ``$`` (e.g., ${VARIABLE_NAME}, where ``{}`` are optional, but safer.)
+
+``pwd`` - Print the current working directory.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  pwd
+  /path/to/directory1
+
+.. note::
+  ``pwd`` is also stored as an environment variable, i.e., ``${PWD}``. Running ``echo ${PWD}`` prints the same output as ``pwd``, but has the advantage of being accessed as part of a file pointer (e.g., ``ls ${PWD}/directory2`` to print all contents of ``directory2`` in the working directory.)
+
+``ls`` - List the contents of a directory.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  ls
+  file1.txt file2.txt directory1
+
+``cd`` - Change the current working directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  cd ~/directory1
+  pwd
+  <$HOME>/directory1
+
+``mkdir`` - Create a new directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+   mkdir directory2
+   ls
+   file1.txt file2.txt directory1 directory2
+
+``rm`` - Remove a file or directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+   rm file1.txt
+   ls
+   file2.txt directory1 directory2
+
+.. note:: 
+  To remove a directory, use ``rmdir`` if the folder is empty. Otherwise, recursively delete the directory and all of its contents via ``rm -r <FOLDER_PATH>``.
+
+``cp`` - Copy a file or directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+   cp file2.txt file3.txt
+   ls
+   file2.txt file3.txt directory1 directory2
+
+.. note:: 
+  Similar to remove, ``cp`` works for files; to copy a folder, along with its contents, it must be done recursively via ``cp -r <FOLDER_PATH> <DESTINATION>``.
+
+``mv`` - Move or rename a file or directory.
+
+.. code-block:: shell
+  :emphasize-lines: 3
+
+  mv file2.txt file4.txt
+  ls
+  file3.txt file4.txt directory1 directory2
+
+``cat`` - Concatenate and display the contents of one or more files.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  cat file3.txt
+  This is the contents of file3.txt
+
+``grep`` - Search for a pattern in a file or input.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  grep "the" file3.txt
+  This is the contents of file3.txt
+
+``sort`` - Sort the lines of a file or input.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  sort file3.txt
+  This is the contents of file3.txt
+
+``uniq`` - Remove duplicates from a sorted file or input.
+
+.. code-block:: shell
+  :emphasize-lines: 2
+
+  sort file3.txt | uniq
+  This is the contents of file3.txt
+
```

Comment:
You might want to include a little text to let people know you are piping the output of the sort into the uniq command because some people might not be familiar with |